loader/loader.go

```diff
@@ -192,12 +218,17 @@ func (w *Worker) restoreDataFile(ctx context.Context, filePath string, offset in
 		return err
 	}
 
-	// dispatchSQL completed, send nil.
+	failpoint.Inject("dispatchError", func(_ failpoint.Value) {
+		w.tctx.L().Info("", zap.String("failpoint", "dispatchError"))
+		failpoint.Return(errors.New("inject failpoint dispatchError"))
+	})
+
+	// dispatchSQL completed, send nil to make sure all dmls are applied to target database
 	// we don't want to close and re-make chan frequently
 	// but if we need to re-call w.run, we need re-make jobQueue chan
 	w.jobQueue <- nil
-
 	w.wg.Wait()
```

Comment:
only `w.wg.Wait` in `Worker.Close` is enough now?

Reply:
It means the file was finished here, the tricky point is that `w.wg` applied on both `file job` and `worker`